includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp-default-footer.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * Service Provider: Mailchimp Default Footer
+ *
+ * @package Newspack
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Usage reports for Mailchimp.
+ */
+class Newspack_Newsletters_Mailchimp_Default_Footer {
+
+	/**
+	 * Initialize hooks.
+	 */
+	public static function init() {
+		add_action( 'admin_footer', [ __CLASS__, 'add_footer_content_to_editor' ] );
+	}
+	/**
+	 * Get the default Mailchimp footer content.
+	 *
+	 * @return string Footer content.
+	 */
+	public static function get_footer_content() {
+
+		$should_append_footer = get_option( 'newspack_mailchimp_auto_append_footer', false );
```

Comment:
Failsafe to bail if the current ESP is no longer Mailchimp:

```suggestion
		$should_append_footer = 'mailchimp' === Newspack_Newsletters::service_provider() && get_option( 'newspack_mailchimp_auto_append_footer', false );
```

Reply:
@copilot